api/urls.py

```diff
@@ -5,4 +5,5 @@
 
 urlpatterns = [
     path('deliveries/', views.DeliveryList.as_view(), name='create'),
+    path('customers/check_is_partner/', views.CustomerList.as_view(), name='check_is_partner'),
```

Comment:
I'm afraid that following an RPC-like approach without defined criteria will become messy pretty soon. I propose instead to implement a REST-like API that listens to the appropriate webhook events and where each endpoint maps to a particular use-case. Then, the REST resources are nouns named after the particular use case.

Our only existing use case so far is the shipment of an order, which is mapped with the `deliveries/` endpoint. Whether or not this requires calling other APIs it's an implementation detail, but from a business perspective is still the shipment of the order. I realize now, that to follow this abstraction a better name would be `create_shipment/` instead. `deliveries` is too coupled to Koiki's API design.

Likewise, for the use case we're implementing here, creating a customer, we could name this `create_customer/` instead. Whether or not this needs to call the CRM's API is an implementation detail that the celery task will deal with. Unfortunately, we miss the HTTP method semantics because webhooks are always sent as POST and I see no other option than including the action in the endpoint's name :shrug: 

That means that we'll have to set up a webhook on Woocommerce for every use case we want to support. So, for example for the case of an update to an order, we would have to create a new webhook with `order.updated` as topic, pointing to a new `update_shipment/` endpoint, which would then deal with Koiki's API.

How does this sound? I'm happy to discuss this further on a call.

Reply:
You are right, in this case since we are restrict to the webhooks payloads we can't identify the "verbs" from the action sent by the webhook and not from the API url itself(making it look RPC). I could just rename it to customers/ similar to 'deliveries' and be consistent with REST.